add yarn workspaces

jroebu14 opened the pull request:
Resolves https://github.com/bbc/psammead/issues/4332

**Overall change:**
Adds [Yarn Workspaces](https://classic.yarnpkg.com/en/docs/workspaces/).

**Update**
Talos depends on Lerna exec so not cannot be removed unless we find a replacement for either Lerna or Talos.

**Code changes:**

- Removes each package's lockfile. Yarn Workspaces uses a single lockfile rather than one for each workspace
- Removes `ci:packages` script:
  - What it did: installed dependencies in each package
  - Why is it removed: because Yarn installs all package's dependencies in a single pass in the root `node_modules`
- Removes `install:packages` script:
  - What it did: installed all package's dependencies and linked any cross-dependencies
  - Why is it removed: because `yarn install` installs all package's dependencies in a single pass in the root `node_modules` and links cross-dependencies by default
- Replaces `lerna exec` with `yarn workspaces run`
  - What it did: runs commands in each package - we use it for building packages with babel
  - Why is it replaced: `yarn workspaces run` is native to Yarn and does the same thing as `lerna exec`
- Changes psammead `moment-timezone-include` package to use commonjs modules. This is because the `postinstall` script in this package's `package.json` (which runs a node script in the build directory) ran before the `postinstall` script in the root `package.json` (which transpiles code and generates the build directories) and attempts to run a script that doesn't yet exist. Using commonjs modules means we don't need to transpile the code before running the node script.
- Updates the `changeScanner` script to check that the root `yarn.lock` has been updated if any package's dependencies have been updated

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

Commit:
replace lerna with yarn workspaces

Commit:
remove redundant workspaces command in test scripts

Commit:
add moment-timezone-include to workspaces nohoist

Commit:
use require over import - removes need for generated script to run

Commit:
remove nohoist packages

Commit:
bump packages

Commit:
cache node modules in GH actions

Commit:
update changelog

Commit:
Revert "cache node modules in GH actions"

This reverts commit 78cd95bbf4858f71ebafd07093fcef71783057c7.

Commit:
remove each package's yarn.lock

Commit:
update changescanner script

chris-hinds commented:
Nice work Jonathan. Does this still allow us to publish packages publicly individually to NPM as we do now. I thought the impression that by default yarn wanted packages to be private?

jroebu14 commented:
I thought it was just the root package that had to be private

chris-hinds commented:
> I thought it was just the root package that had to be private

You are right, just read up on it again.

andrewscfc commented on scripts/changeScanner/index.js:
```diff
@@ -14,28 +14,24 @@ const requiredChanges = ['CHANGELOG.md', packageFileName];
-    devDependencies: remoteDevDeps,
-  } = JSON.parse(remotePackageFile);
-  const depsHaveChanged = !equals(localDeps, remoteDeps);
-  const devDepsHaveChanged = !equals(localDevDeps, remoteDevDeps);
```
Not sure I follow the changes in this file; why are devDeps no longer checked?

jroebu14 replied:
They don't affect the root yarn.lock

andrewscfc commented on scripts/changeScanner/index.js:
```diff
@@ -14,28 +14,24 @@ const requiredChanges = ['CHANGELOG.md', packageFileName];
+
+  return packageChanges
+    ? packageChanges.find(changedFile => changedFile.includes(matchFile))
+    : '';
```
What has changed to make `packageChanges` sometimes `undefined`?

jroebu14 replied:
this change https://github.com/bbc/psammead/pull/4368/files#diff-b91241351ff6aaef3a16d76228a999a9994e7038269586f10e5eb64e2b1d9657R45-R48

This is selecting the psammead root package which doesn't exist in `changedPackages` if there haven't been any changes in psammead root

andrewscfc reviewed (approved):
LGTM, really excited to see if this facilitates cross repo dev 🤞 Just a few questions around the change scanner but not blocking

chris-hinds reviewed (approved):
Changes look great. Thanks for this @jroebu14

FK78 committed:
Merge branch 'latest' into replace-lerna-with-yarn-workspaces

Commit:
Merge branch 'latest' into replace-lerna-with-yarn-workspaces

Commit:
remove spaces in chromaticqa action

andrewscfc committed:
prevent build from putting test files in build output

andrewscfc committed:
fix moment version in psammead-locale

Commit:
fix moment versions in psammead-timestamp-container

Commit:
fix psammead-section-label test

Commit:
reintroduce lerna

Commit:
remove unused script

Commit:
Bump package versions

Bump package versions for psammead, psammead-amp-geo, psammead-brand, psammead-bulleted-list, psammead-bulletin, psammead-byline, psammead-caption, psammead-consent-banner, psammead-content-anchor, psammead-copyright, psammead-embed-error, psammead-episode-list, psammead-figure, psammead-grid, psammead-heading-index, psammead-headings, psammead-image-placeholder, psammead-image, psammead-inline-link, psammead-live-label, psammead-media-indicator, psammead-media-player, psammead-most-read, psammead-navigation, psammead-oembed, psammead-paragraph, psammead-play-button, psammead-podcast-promo, psammead-radio-schedule, psammead-script-link, psammead-section-label, psammead-sitewide-links, psammead-social-embed, psammead-story-promo-list, psammead-story-promo, psammead-timestamp, psammead-topic-tags, psammead-useful-links, psammead-visually-hidden-text, psammead-timestamp-container, gel-foundations, moment-timezone-include, psammead-assets, psammead-calendars, psammead-detokeniser, psammead-locales, psammead-rich-text-transforms, psammead-storybook-helpers, psammead-styles, psammead-test-helpers, web-vitals

Commit:
Update changelogs

Updates changelogs for psammead, psammead-amp-geo, psammead-brand, psammead-bulleted-list, psammead-bulletin, psammead-byline, psammead-caption, psammead-consent-banner, psammead-content-anchor, psammead-copyright, psammead-embed-error, psammead-episode-list, psammead-figure, psammead-grid, psammead-heading-index, psammead-headings, psammead-image-placeholder, psammead-image, psammead-inline-link, psammead-live-label, psammead-media-indicator, psammead-media-player, psammead-most-read, psammead-navigation, psammead-oembed, psammead-paragraph, psammead-play-button, psammead-podcast-promo, psammead-radio-schedule, psammead-script-link, psammead-section-label, psammead-sitewide-links, psammead-social-embed, psammead-story-promo-list, psammead-story-promo, psammead-timestamp, psammead-topic-tags, psammead-useful-links, psammead-visually-hidden-text, psammead-timestamp-container, gel-foundations, moment-timezone-include, psammead-assets, psammead-calendars, psammead-detokeniser, psammead-locales, psammead-rich-text-transforms, psammead-storybook-helpers, psammead-styles, psammead-test-helpers, web-vitals

Commit:
merge latest

Commit:
Merge branch 'latest' into replace-lerna-with-yarn-workspaces

Commit:
trigger build

Commit:
trigger build

Commit:
unbump moment-timezone-include

Commit:
update docs

Commit:
Merge branch 'latest' into replace-lerna-with-yarn-workspaces

Commit:
bump psammead

Commit:
Merge branch 'latest' into replace-lerna-with-yarn-workspaces
